### ISSUE TITLE
Multiple Interfaces -working!

### ISSUE THREAD

**Issue by richtj999** [jokob-sk/Pi.Alert#83]
Hi,

Thanks for getting the multiple interfaces working!  I have it working within a Ubuntu VM running docker nested.

I had a couple questions though:

1.  The changes I make to the Pialert dont seem to survive a reboot of the vm.  This might be a user error (mine).
2.  The IP for the docker pialert access is 172.17.0.1:20211 but I am unable to access that if i am not on the host machine.  
3.  If i went the non-docker route - is there a fork with your updates without it being a docker file?

This is exactly what I am looking for - thanks for working on it!  

Thank you,
Rich

**Comment by jokob-sk** [comment]
Hi Rich!

Thanks for checking out this fork/container!

1) Regrettably I'm not familiar with VMs so I can't help here. :/ I guess it's some generic VM thing, AFAIK, for data persistence in this container you only need the `pialert.db` file and `pialert.conf`. 
2) This image needs to be run on the host network (see [docker-compose config examples](https://github.com/jokob-sk/Pi.Alert/tree/main/dockerfiles) which might help) so maybe that's what's causing this. Again, might be VM specific so not 100% sure. 
3) Regrettably I'm unaware of one. I prefer environment-agnostic deploys, so focusing on Docker as I'm not enjoying solving HW/environment specific issues :)

Sorry for not being of much help here.

j.

**Comment by richtj999** [comment]
OK - Im sticking with Docker!  I need a few tips / help is possible?

My process now

Sudo apt-get update
Sudo apt-get Install docker
Sudo docker pull jokobsk/pi.alert
sudo docker run --rm --net=host jokobsk/pi.alert

from there I am up and running at 172.17.0.1:20211 – only on the host machine.

After a reboot of the ubuntu host, the pialert docker doesn’t turn back on

I can use:  docker exec -u 0 -it 8662ea2fa000 /bin/bash

Install nano, edit the config file in the docker container, add the two interfaces and have it working

A few questions which are docker related – im happy to use docker if I can figure these few things out.  

1.	 Can I access Pialert on a machine beyond the docker PC?  The 172.17.0.1 network is unique to that host - My main network im using is 192.168.80.0/24 – interface ens160.  
a.	IE – if the host has a docker IP of 172.17.0.1 and a ‘real’ IP of 192.168.80.25 – can PiAlert be accessed on both?
b.	If not – how are people accessing pialert when not on the local host machine?
2.	Im a little fuzzy on the exact command to make /home/pi/pialert/config to a persistent folder.  
a.	If my physical pc is /home/richtj999/Documents/dockerpersistent – what command do I use to let the container know that’s what I want (volume command but not sure how to make it happen).
b.	I assume once the persistent config & db are in place, it works perfect!
3.	  Im really new to docker and still am not getting the docker-compose.yml (https://github.com/jokob-sk/Pi.Alert/tree/main/dockerfiles#-examples) 
a.	I get how its supposed to work but I am not understanding how to ‘run’ it.  
b.	I installed the sudo docker-compose and sort of had it working but I think im building images, not loading them, not able to start them.


TLDR:  
Can you help me:
1.	Autostart on reboot of host PC
2.	Set persistent storage for config/db?
3.	Add ability to access pialert from local network (192.168.80.25)?

Thanks again,
Rich

**Comment by jokob-sk** [comment]
Hi Rich!

I'd love to help, but again, VMs are not something I worked with. With my limited knowledge, here is what I recommend and I probably can't dive too much deeper than this advice :/

To your questions:

1) I'm not sure as I haven't worked with VMs, however I'm outlining some possibilities below. This is a docker and VM networking issue so I assume any community (reddit/SO?) with such focus might be able to help.
2) The persistence of docker containers is achieved via file system mapping: http://sefidian.com/2021/11/05/volumes-in-docker-compose-tutorial/
3) Use the sample docker-compose and env files provided here: 
https://github.com/jokob-sk/Pi.Alert/blob/main/.env
https://github.com/jokob-sk/Pi.Alert/blob/main/docker-compose.yml
and run it as per the readme in example 1: https://github.com/jokob-sk/Pi.Alert/tree/main/dockerfiles#example-1

I assume, the process would be something like:

1) I strongly recommend to streamline the docker startup with docker-compose (see the aforementioned examples in the documentation)
2) Map the DB and conf files to have a persistent state for your container
3) Test that you can destroy and spin up the container with docker-compose (see the volumes: section in the docker-compose file/examples)
4) Give the container (and I assume subsequently the VM) access to the network
5) Write  abash script to launch your docker instance 
6) Execute the above script on VM boot  (via cron or system.d maybe)

This should give you a repeatable and robust process.

hope this helps,
j.

**Comment by richtj999** [comment]
Hey just wanted to thank you again for all the tips above - I have it all working!  Im really happy!

I didnt realize I needed to edit the docker-compose.yml file for my own filesystem.

`    volumes:
      - ${APP_DATA_LOCATION}/pialert/config:/home/pi/pialert/config`
 Change to:
`     - /home/rich/pi/pialert/config:/home/pi/pialert/config'`

The networking was correct the whole time - i just didnt go to the right IP (duh)

`sudo docker-compose up` was all I needed to do.  

This is great - again thank you so much!

**Comment by richtj999** [comment]
Sorry  - not sure how to close this but again thanks!

**Comment by jokob-sk** [comment]
Ah, glad to hear that, your message made my morning :)
Happy docker journey! 🐳 (I think containers are neat :+1: )
